cognite/client/data_classes/assets.py

```diff
@@ -259,7 +259,10 @@ def _retrieve_related_resources(self, resource_list_class, resource_api):
             resource_api.list, tasks, resource_api._config.max_workers
         ).results
         resources = resource_list_class([])
+        file_ids = []
         for res in res_list:
+            res = resource_list_class([item for item in res if item.id not in file_ids])
+            file_ids += [item.id for item in res]
```

Comment:
Actually, it turns out that this will not work; the set does not support the += operation. But it does have a set.union function. So I would have to do something like:
seen_file_ids = seen_file_ids.union({item.id for item in res}).
But this is perhaps less clear than the original expression and uses the imperative programming style. I don't really prefer one over the other.

Reply:
Alright, let's keep it simple and stick with lists then. :+1: